### DISCUSSION TITLE
RSI warning icon

### DISCUSSION THREAD

**Discussion by RaiLTC** [ppy/osu#12997] [Ideas]
Didn't even know why I had pain before I had to learn why the hard way.
And chance to add an icon which if you hover over it will display a message regarding RSI?
It will say something like "Playing for a long period of time may cause strain on hands and lead to RSI".
Not so great with words but you can figure out something if you plan on adding it. Really important IMO.

**Comment by Aergwyn** [comment]
As long as I have the option to disable it I'm all for it.

**Comment by RaiLTC** [comment]
Just like the tips on the main menu. Hide the icon if desired!

**Comment by goatdavid** [comment]
i back this. kind of like on the Wii, whenever you were playing for an
extended period of time, they'd tell you to get some sunlight... lazer
could do the same in the fact that if someone is playing for like about an
hour+, they'll get a non-intrusive reminder that RSI exists...

**Comment by RaiLTC** [comment]
A message about taking a break is not good. Need to say RSI straight up or people will just ignore it without being aware of the actual problem repetitive movement can cause.

Maybe even have the icon link to the wiki of RSI.

**Comment by Tiddons** [comment]
I am all for the idea. But how long would it take to bring up the icon?

**Comment by LovelyA72** [comment]
How about add the RSI warning to the splash screen? The screen that we are using to display ''this is a development version''. Because I saw a lot of commercial videogames use that space for photosensitive epileptic  seizure warning.

**Comment by goatdavid** [comment]
that's an option, but I'm not sure its effectiveness would be what we're
hoping for

**Comment by Tiddons** [comment]
That wont really warn them while there playing. and placing it if you start any song will annoy people.

**Comment by LovelyA72** [comment]
@Vsyncrooo I mean... This one...
![image](https://user-images.githubusercontent.com/29054052/45259823-d0847300-b38a-11e8-88e2-fceb298b329c.png)

**Comment by Tiddons** [comment]
yea i know

**Comment by Tiddons** [comment]
ofcourse it could be added but i think we need a way to let people know while they are in the menu without annoying the player. a slash screen can be forgotten/ignored easely .

**Comment by Alex4386** [comment]
Instead of Splash Screen, How about just using in-game notification thing?

E.g.
```
Warning Icon. You are playing for 00 hours.
```

**Comment by peppy** [comment]
If someone wishes to attempt to implement this then please ask for a design.

**Comment by Xinnoh** [comment]
different idea, but the warning could suggest playing a different mode
since most people suck at playing the other modes it wouldn't be strenuous + is more appealing than taking a break

**Comment by Tiddons** [comment]
Thats not a bad idea. But then we would still have a problem on how to warn people.

**Comment by goatdavid** [comment]
how about this:

when the user has had osu! open for at least an hour, and is actively playing, when they select a song from song select, the warning dialogue will show up. it could be like the message that comes up when osu! detects a missing background in a beatmap, but it could literally be something as simple as "you've been playing osu!standard for about an hour now, consider taking a break or trying a different mode!"

idk

**Comment by Tiddons** [comment]
Would a icon not be better. Unless the warning isnt that annoying.

**Comment by RaiLTC** [comment]
It doesn't have to be a message or anything. It's as simple as I said, just a warning icon (those usual yellow triangle-shaped icons with an exclamation mark inside it could do the work). Hover your mouse over it to display the desired warning message.
And you don't need it to cover half of the screen, just find some space which you can see it clearly (I believe a good place would be either top right of the screen or left to the mods icon)

It doesn't have to be something that will annoy players every hour, so having the option to hide it in the settings is great.

We only need to let players be aware of RSI and what it can cause.. Nothing more than that.
Plus it can remove some of the "my hand hurts what should I do" messages from the forums.

**Comment by peppy** [comment]
Just to make myself clear: this conversation does not need to continue. If someone wishes to attempt implementation of this, please ask for a design.

**Comment by EVAST9919** [comment]
Please, provide the design for this

**Comment by arflyte** [comment]
Need copy. I'll just design an empty box for now.

**Comment by UselessToucan** [comment]
What about the design?

**Comment by recognitionmark** [comment]
All of your ideas are good, I have to say. But what if we allow a mapper to enable these in their maps individually (as they did with the photosensitive epilepsy warning) so e.g. Sotarks is going to map Raise My Sword, but when you play the map, it will tell you a message of "playing this map for too long can cause RSI because of (insert cause here, whether jumps or streams or etc.). Taking frequent breaks will help."

**Comment by recognitionmark** [comment]
Maybe the design would be a stream and an exclamation mark? The design of course should show something that will damage the hand.

**Comment by peppy** [comment]
it would not be a beatmap level toggle, no.

**Comment by lilyremigia** [comment]
I have seen two main design ideas pop-up here: a warning at the startup of the game and a periodical reminder.
These two should be separated into their own issues, no?
As for the periodical reminder: the time, and method should probably be user configurable.
Also, the simplest method of that would probably be just a modal, that also waits for the end of the play, if the time is up, in a session.

**Comment by bdach** [comment]
No, they should be not separated into two issues. And I don't think the period should be configurable. Nobody will use that setting.

**Comment by peppy** [comment]
This would not be configurable and would be incorporated with the intro / startup design in a tasteful way if it is to be implemented. No further discussion is required; please see my [previous comment](https://github.com/ppy/osu/issues/3352#issuecomment-419693949) for the correct direction forward.
